### PR TITLE
[YUNIKORN-1935] add gang scheduling with priority test

### PR DESCRIPTION
**Pull request by FrankYang0529**
### What is this PR for?

Add gang scheduling with PriorityClass e2e test case.

### What type of PR is it?
* [X] - Improvement

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-1935

### CODE REVIEW

**Review by manirajv06 — CHANGES_REQUESTED** [review]
looks good.

Minor changes

File: test/e2e/priority_scheduling/priority_scheduling_test.go [hunk 1]
```diff
@@ -251,6 +252,217 @@ var _ = ginkgo.Describe("PriorityScheduling", func() {
...
+						Name:        "group-high",
+						MinMember:   int32(1),
+						MinResource: taskGroupMinResource,
+					},
+				},
+			},
+			Resources:         rr,
+			PriorityClassName: highPriorityClass.Name,
+			RestartPolicy:     v1.RestartPolicyNever,
+		}
```
Comment (manirajv06) [inline test/e2e/priority_scheduling/priority_scheduling_test.go:346]: See if you can make the whole code starting from line no.285 simpler by adding a new method to return the test pod config for all high, low etc

File: test/e2e/priority_scheduling/priority_scheduling_test.go [hunk 2]
```diff
@@ -251,6 +252,217 @@ var _ = ginkgo.Describe("PriorityScheduling", func() {
...
+		Ω(err).NotTo(gomega.HaveOccurred())
+
+		By("Submit high priority job")
+		highPod, err = k8s.InitTestPod(highPodConf)
+		Ω(err).NotTo(gomega.HaveOccurred())
+		highJob := k8s.InitTestJob(highPod.Labels[constants.LabelApplicationID], 1, 1, highPod)
+		highJob, err = kubeClient.CreateJob(highJob, ns)
+		Ω(err).NotTo(gomega.HaveOccurred())
+		err = kubeClient.WaitForJobPodsCreated(ns, highJob.Name, 1, 30*time.Second)
+		Ω(err).NotTo(gomega.HaveOccurred())
```
Comment (manirajv06) [inline test/e2e/priority_scheduling/priority_scheduling_test.go:382]: Can we submit the jobs in reverse order?

**Review by manirajv06 — CHANGES_REQUESTED** [review]

File: test/e2e/priority_scheduling/priority_scheduling_test.go (hunk 2 above)
Comment (FrankYang0529) [reply test/e2e/priority_scheduling/priority_scheduling_test.go:325]: I would like to submit from low priority job, so we can test the system consider priority first and then submission time.

**Pull request closed without merging**